src/runtime_src/core/pcie/driver/linux/xocl/xocl_subdev.c

```diff
@@ -1001,6 +1022,26 @@ void xocl_subdev_destroy_by_level(xdev_handle_t xdev_hdl, int level)
 	xocl_unlock_xdev(xdev_hdl);
 }
 
+void xocl_subdev_destroy_by_slot(xdev_handle_t xdev_hdl,
+				 uint32_t slot_id)
+{
+	struct xocl_dev_core *core = (struct xocl_dev_core *)xdev_hdl;
+	int i, j;
+
+	xocl_lock_xdev(xdev_hdl);
+	for (i = ARRAY_SIZE(core->subdevs) - 1; i >= 0; i--)
```

Comment:
Scary piece of code.
While the probability is near zero, think about what happens if the ARRAY_SIZE is zero.

Is there a reason why the code needs to count down.  Counting up looks like it would work AND the code isn't as scary.

Reply:
subdevs is a static array hence size will not be 0. But you are correct, it could be written better way. As this is followed by an existing code in this file, I believe, we should not change much now.